File: JavaScript/CoinBlock.js
class CoinBlock extends Sprite{
    constructor(x, y, model){
        super(x, y, 50, 50, model);

        this.image = new Image();
        this.image.src = "images/coinBlock.png";

        this.depletedImage = new Image();
        this.depletedImage.src = "images/depletedCoinBlock.png";

        this.blockHit = false;
        this.coinCount = 5;
        this.isCoinBlock = true;
    }


    update(){   }

    draw(ctx){
        if(this.coinCount > 0)
            ctx.drawImage(this.image, this.x - this.model.screenPos, this.y);
        else    
            ctx.drawImage(this.depletedImage, this.x - this.model.screenPos, this.y);
    }

    ejectCoin(){
        if(this.coinCount > 0){
            const tempCoin = new Coin(this.x, this.y, this.model);
            this.model.sprites.push(tempCoin);
            this.coinCount--;
        }else{
            console.log("coin Block is out of coins");
        }
    }
}

File: Python/CoinBlock.py
import pygame

from Sprite import *
from Coin import *

class CoinBlock(Sprite):

    def __init__(self, model, x, y, w, h):
        super(CoinBlock, self).__init__(x,y,w,h)  
        self.w = 50
        self.h = 50
        self.model = model
        self.coins = 5
        self.imageIndex = 0
        self.coinBlockImages = []
        self.lazyLoad()

        isCoinBlock = True

    def update(self):
        if self.coins < 1:
            self.imageIndex = 1
        else:
            self.imageIndex = 0

    def draw(self, screen):
        screen.blit(self.coinBlockImages[self.imageIndex], (self.x - self.model.screenPos, self.y))

    def ejectCoin(self):
        tempCoin = Coin(self.model, (self.x + 4), self.y, 30 ,30)
        self.model.sprites.insert(len(self.model.sprites), tempCoin)
        self.coins -= 1

    def lazyLoad(self):
        self.coinBlockImages = [
            pygame.image.load("images/coinBlock.png"), 
            pygame.image.load("images/depletedCoinBlock.png")
        ]
